src/Team-Services-Bot.Api/Dialogs/ConnectDialog.cs

```diff
@@ -65,8 +64,15 @@ public Task StartAsync(IDialogContext context)
 
         private static string GeneratePin()
         {
-            var generator = new Random();
-            return generator.Next(Min, Max).ToString("0000", CultureInfo.InvariantCulture);
+            using (var generator = new RNGCryptoServiceProvider())
+            {
+                var data = new byte[4];
+
+                generator.GetBytes(data);
+                var value = BitConverter.ToUInt16(data, 0);
```

Comment:
Basically, it's private method. So it can be (should be) tested indirectly through the public method calls.